plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx

```diff
@@ -80,22 +84,27 @@ export const EntitySwitch = (props: EntitySwitchProps) => {
             return [];
           }
 
-          const { if: condition, children: elementsChildren } =
-            element.props as EntitySwitchCase;
+          const {
+            if: condition,
+            children: elementsChildren,
+            default: isDefault,
+          } = element.props as EntitySwitchCase;
 
           // If the entity is missing or there is an error, render the default page
           if (!entity) {
             return [
               {
                 if: condition === undefined,
                 children: elementsChildren,
+                default: isDefault ?? false,
```

Comment:
Yeah I see that it'll require a little bit of internal refactoring here to support what we want to do. Maybe you'll even want to remove the flatMap and upfront decide what the default case is based on `if` absence (bonus points for detecting that there's more than one and logging a warning?), and perhaps only lazily evaluating their `if` statements in a `for` loop, so that you never execute more than the necessary number of `if` callbacks if they happen to be expensive functions? At least in the `'first'` case, of course.

Reply:
Yeah, there might be a place for optimization. Actually, it should have been done already in the beginning. But for my Friday afternoon brain, this was a too hard task to go with so I just refactored the code by the above comments. Maybe something to think about later as it will change the whole file dramatically.